ddl/table.go

```diff
@@ -356,6 +356,31 @@ func (d *ddl) onModifyTableComment(t *meta.Meta, job *model.Job) (ver int64, _ e
 	return ver, nil
 }
 
+func (d *ddl) onRenameIndex(t *meta.Meta, job *model.Job) (ver int64, _ error) {
+	var from, to model.CIStr
+	if err := job.DecodeArgs(&from, &to); err != nil {
+		job.State = model.JobStateCancelled
+		return ver, errors.Trace(err)
+	}
+
+	tblInfo, err := getTableInfo(t, job, job.SchemaID)
+	if err != nil {
+		return ver, errors.Trace(err)
+	}
+
+	idx := findIndexByName(from.L, tblInfo.Indices)
```

Comment:
I'm afraid we have to double check here if we don't take @coocood 's  suggestion.
Take this scenario for example:
TiDB client1,  rename idx1 to idx3,   check success,  put into queue
TiDB client2,  rename idx2 to idx3,   check success,  put into queue

When the code runs to here, those two request would both success! 
We lost a index, no error!

Reply:
yes, double check is needed here.